code/modules/multiz/hoist.dm

```diff
@@ -0,0 +1,210 @@
+///////////////////////////
+// Dost thou even hoist? //
+///////////////////////////
+
+/obj/item/hoist_kit
+	name = "hoist kit"
+	desc = "A setup kit for a hoist that can be used to lift things. The hoist will deploy in the direction you're facing."
+	icon = 'icons/obj/hoists.dmi'
+	icon_state = "hoist_case"
+
+/obj/item/hoist_kit/attack_self(mob/user)
+	new /obj/structure/hoist (get_turf(user), user.dir)
+	user.visible_message(span("warning", "[user] deploys the hoist kit!"), span("notice", "You deploy the hoist kit!"), span("notice", "You hear the sound of parts snapping into place."))
+	qdel(src)
+
+/obj/effect/hoist_hook
+	name = "hoist clamp"
+	desc = "A clamp used to lift people or things."
+	icon = 'icons/obj/hoists.dmi'
+	icon_state = "hoist_hook"
+	var/obj/structure/hoist/source_hoist
+	anchored = 1
+
+/obj/effect/hoist_hook/MouseDrop_T(atom/movable/AM,mob/user)
+	if(!istype(AM))
+		log_debug("AAAAAAA")
+		return
+	if (!AM.simulated || AM.anchored || source_hoist.hoistee || !Adjacent(AM))
```

Comment:
Some better error messaging would be nice. For the 1st, 2nd, and 4th condition, `You can't do this!` would be appropriate. For the 3rd case, `[source_hoist.hoistee] is already buckled to [src]!`